File: lemur-libs/lemur-contract/src/main/java/io/github/malczuuu/lemur/contract/message/player/PlayerCreatedMessage.java
package io.github.malczuuu.lemur.contract.message.player;

import com.fasterxml.jackson.annotation.JsonIgnore;
import io.github.malczuuu.lemur.contract.TransportMessage;

public record PlayerCreatedMessage(String playerId) implements PlayerMessage, TransportMessage {

  @JsonIgnore
  @Override
  public String messageType() {
    return "PlayerCreated";
  }
}

File: lemur-libs/lemur-contract/src/main/kotlin/io/github/malczuuu/lemur/contract/message/player/PlayerCreatedMessage.kt
package io.github.malczuuu.lemur.contract.message.player

import io.github.malczuuu.lemur.contract.TransportMessage
import jakarta.validation.constraints.NotNull

/**
 * See [TransportMessage] for explanation of why fields are nullable.
 */
data class PlayerCreatedMessage(@field:NotNull val playerId: String?) : TransportMessage
